### PR TITLE
Rect collidelistall unittests added and updated docs to document different argument types

### PR DESCRIPTION
**Pull request by dr0id**
Reading the collidelistall method in c and then the unittests I found that there are some missing to fully test the c code paths. Therefor added more tests with the different types of arguments.

Also updated the docs to document the different argument types.

### CODE REVIEW

File: docs/reST/ref/rect.rst [hunk 1]
```diff
@@ -348,6 +348,52 @@
...
+          list_of_lists = [[1,1,1,1], [2,2,2,2]]
+          indices1 = r.collidelistall(list_of_lists)
+    
+          list_of_tuples = [(1,1,1,1), (2,2,2,2)]
+          indices2 = r.collidelistall(list_of_tuples)
+    
+          list_of_double_tuples = [((1,1), (1,1)), ((2,2), (2,2))]
+          indices3 = r.collidelistall(list_of_double_tuples)
+    
+          class ObjectWithRectAttribute(object):
```
Comment (novialriptide) [inline docs/reST/ref/rect.rst:371]: You could remove the (object) here because of Python3. Same for the others

**Review by novialriptide — CHANGES_REQUESTED** [review]

File: docs/reST/ref/rect.rst (hunk 1 above)
Comment (dr0id) [reply docs/reST/ref/rect.rst:371]: sure, old habits, thanks :)

File: docs/reST/ref/rect.rst [hunk 2]
```diff
@@ -348,6 +348,52 @@
...
+                  self._rect = r
+    
+              def rect(self):
+                  return self._rect
+    
+          list_of_object_with_callable_rect = [
+                                              ObjectWithCallableRectAttribute(Rect(1,1,1,1)), 
+                                              ObjectWithCallableRectAttribute(Rect(2,2,2,2))
+          ]
+          indices5 = r.collidelistall(list_of_object_with_callable_rect)
```
Comment (ankith26) [inline docs/reST/ref/rect.rst:394]: minor nitpick, this codeblock could do better with `black` formatting. We use black throughout the codebase, so as to maintain consistency ;)

Comment (dr0id) [reply docs/reST/ref/rect.rst:392]: your are right, reformatted example code with black

**Review by MyreMylar — APPROVED** [review]
LGTM. Can't hurt to have some more valid tests with lots of Rect changes being worked on.

**Review by illume — APPROVED** [review]
👍 🎈 thanks

**Pull request merged**